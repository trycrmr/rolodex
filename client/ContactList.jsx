import React from "react";
import { Box, Columns } from "react-bulma-components";
import ContactCard from "./ContactCard.jsx";
const ContactList = (props) => {
  return (
    <>
      <Box>
        <Columns breakpoint="tablet">
          <Columns.Column>
            <ContactCard />
          </Columns.Column>
          <Columns.Column>
            <ContactCard />
          </Columns.Column>
        </Columns>
        <Columns breakpoint="tablet">
          <Columns.Column>
            <ContactCard />
          </Columns.Column>
          <Columns.Column>
            <ContactCard />
          </Columns.Column>
        </Columns>
      </Box>
    </>
  );
};

export default ContactList;
