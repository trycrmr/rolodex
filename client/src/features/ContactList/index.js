import React from "react";
import { Box, Columns, Heading } from "react-bulma-components";
import ContactCard from "../../ContactCard.jsx";
import { Link } from "react-router-dom";
const ContactList = (props) => {
  // I was going for a "business card" look for this contacts with the bulma react components :-) .
  return (
    <>
      <Box style={{ margin: "auto" }}>
        {props.data.length === 0 ? (
          <Heading style={{ textAlign: "center" }}>
            You don't have any contacts; Why not <Link to="/add">add one</Link>?
          </Heading>
        ) : (
          props.data.map((thisContact, idx, origArr) => {
            if (idx % 2) {
              // Only process the even indexes so we can have two cards per column
              return null;
            } else {
              return (
                // I wasn't sure what keys to use here. Math.random is non-deterministic and using the index is frowned upon, but really the identifying component is the Contact card, so I used the contact.id as the key there. Probably could do this better, but idk what bugs with the current setup.
                <Columns
                  breakpoint="tablet"
                  key={Math.random()}
                  style={{ backgroundColor: "whitesmoke" }}
                >
                  <Columns.Column key={idx}>
                    <ContactCard data={thisContact} key={thisContact.id} />
                  </Columns.Column>
                  {origArr[idx + 1] ? (
                    <Columns.Column key={idx + 1}>
                      <ContactCard
                        data={origArr[idx + 1]}
                        key={origArr[idx + 1].id}
                      />
                    </Columns.Column>
                  ) : null}
                </Columns>
              );
            }
          })
        )}
      </Box>
    </>
  );
};

export default ContactList;
