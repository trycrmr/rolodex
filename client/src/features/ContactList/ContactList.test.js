import React from "react";
import renderer from "react-test-renderer";
import ContactList from "../ContactList";
import { MemoryRouter } from "react-router-dom";

const mockData = [
  {
    name: "Milan Jenkins",
    email: "Kristian_Deckow@example.org",
    phone: "330-013-0169",
    id: "e12970f9-1deb-48a2-a4c2-854d86a82ef0",
  },
  {
    name: "Dr. Fletcher Doyle",
    email: "Bernadine_Mitchell@example.net",
    phone: "213-585-2178",
    id: "1e580e71-c00e-4dad-9692-bbe1b6e93e37",
  },
];

describe("The Contact List component", () => {
  test("renders properly with no contacts", () => {
    expect(
      renderer
        .create(
          <MemoryRouter>
            <ContactList data={[]} />
          </MemoryRouter>
        )
        .toJSON()
    ).toMatchSnapshot();
  });

  test("renders properly with one contact", () => {
    expect(
      renderer
        .create(
          <MemoryRouter>
            <ContactList data={[mockData[0]]} />
          </MemoryRouter>
        )
        .toJSON()
    ).toMatchSnapshot();
  });

  test("renders properly with two contacts", () => {
    expect(
      renderer
        .create(
          <MemoryRouter>
            <ContactList data={mockData} />
          </MemoryRouter>
        )
        .toJSON()
    ).toMatchSnapshot();
  });
});
