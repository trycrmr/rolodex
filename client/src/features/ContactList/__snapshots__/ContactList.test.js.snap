// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly with no contacts 1`] = `
<div
  className="box"
  style={
    Object {
      "margin": "auto",
    }
  }
>
  <h1
    className="title"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    You don't have any contacts; Why not 
    <a
      href="/add"
      onClick={[Function]}
    >
      add one
    </a>
    ?
  </h1>
</div>
`;

exports[`renders properly with one contact 1`] = `
<div
  className="box"
  style={
    Object {
      "margin": "auto",
    }
  }
>
  <div
    className="columns is-tablet is-multiline"
    style={
      Object {
        "backgroundColor": "whitesmoke",
      }
    }
  >
    <div
      className="column"
    >
      <div
        className="card"
        style={
          Object {
            "margin": "auto",
            "maxWidth": "500px",
          }
        }
      >
        <div
          className="card-content"
        >
          <article
            className="media"
            position="right"
          >
            <figure
              className="media-left"
            >
              <figure
                className="image is-128x128"
              >
                <img
                  alt="128x128"
                  className=""
                  onError={[Function]}
                  src="http://bulma.io/images/placeholders/128x128.png"
                />
              </figure>
            </figure>
            <div
              className="media-content"
              style={
                Object {
                  "margin": "auto",
                  "textAlign": "center",
                }
              }
            >
              <h1
                className="title is-6"
              >
                Milan Jenkins
              </h1>
              <div
                className="content"
              >
                Kristian_Deckow@example.org
                <br />
                330-013-0169
              </div>
            </div>
          </article>
        </div>
        <div
          className="card-footer"
        >
          <div
            className="card-footer-item"
            href="/view"
          >
            <a
              href="/view/e12970f9-1deb-48a2-a4c2-854d86a82ef0"
              onClick={[Function]}
            >
              View
            </a>
          </div>
          <div
            className="card-footer-item"
            href="/edit"
          >
            <a
              href="/edit/e12970f9-1deb-48a2-a4c2-854d86a82ef0"
              onClick={[Function]}
            >
              Edit
            </a>
          </div>
          <div
            className="card-footer-item"
            href="/delete"
          >
            <a
              href="/delete/e12970f9-1deb-48a2-a4c2-854d86a82ef0"
              onClick={[Function]}
            >
              Delete
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders properly with two contacts 1`] = `
<div
  className="box"
  style={
    Object {
      "margin": "auto",
    }
  }
>
  <div
    className="columns is-tablet is-multiline"
    style={
      Object {
        "backgroundColor": "whitesmoke",
      }
    }
  >
    <div
      className="column"
    >
      <div
        className="card"
        style={
          Object {
            "margin": "auto",
            "maxWidth": "500px",
          }
        }
      >
        <div
          className="card-content"
        >
          <article
            className="media"
            position="right"
          >
            <figure
              className="media-left"
            >
              <figure
                className="image is-128x128"
              >
                <img
                  alt="128x128"
                  className=""
                  onError={[Function]}
                  src="http://bulma.io/images/placeholders/128x128.png"
                />
              </figure>
            </figure>
            <div
              className="media-content"
              style={
                Object {
                  "margin": "auto",
                  "textAlign": "center",
                }
              }
            >
              <h1
                className="title is-6"
              >
                Milan Jenkins
              </h1>
              <div
                className="content"
              >
                Kristian_Deckow@example.org
                <br />
                330-013-0169
              </div>
            </div>
          </article>
        </div>
        <div
          className="card-footer"
        >
          <div
            className="card-footer-item"
            href="/view"
          >
            <a
              href="/view/e12970f9-1deb-48a2-a4c2-854d86a82ef0"
              onClick={[Function]}
            >
              View
            </a>
          </div>
          <div
            className="card-footer-item"
            href="/edit"
          >
            <a
              href="/edit/e12970f9-1deb-48a2-a4c2-854d86a82ef0"
              onClick={[Function]}
            >
              Edit
            </a>
          </div>
          <div
            className="card-footer-item"
            href="/delete"
          >
            <a
              href="/delete/e12970f9-1deb-48a2-a4c2-854d86a82ef0"
              onClick={[Function]}
            >
              Delete
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="column"
    >
      <div
        className="card"
        style={
          Object {
            "margin": "auto",
            "maxWidth": "500px",
          }
        }
      >
        <div
          className="card-content"
        >
          <article
            className="media"
            position="right"
          >
            <figure
              className="media-left"
            >
              <figure
                className="image is-128x128"
              >
                <img
                  alt="128x128"
                  className=""
                  onError={[Function]}
                  src="http://bulma.io/images/placeholders/128x128.png"
                />
              </figure>
            </figure>
            <div
              className="media-content"
              style={
                Object {
                  "margin": "auto",
                  "textAlign": "center",
                }
              }
            >
              <h1
                className="title is-6"
              >
                Dr. Fletcher Doyle
              </h1>
              <div
                className="content"
              >
                Bernadine_Mitchell@example.net
                <br />
                213-585-2178
              </div>
            </div>
          </article>
        </div>
        <div
          className="card-footer"
        >
          <div
            className="card-footer-item"
            href="/view"
          >
            <a
              href="/view/1e580e71-c00e-4dad-9692-bbe1b6e93e37"
              onClick={[Function]}
            >
              View
            </a>
          </div>
          <div
            className="card-footer-item"
            href="/edit"
          >
            <a
              href="/edit/1e580e71-c00e-4dad-9692-bbe1b6e93e37"
              onClick={[Function]}
            >
              Edit
            </a>
          </div>
          <div
            className="card-footer-item"
            href="/delete"
          >
            <a
              href="/delete/1e580e71-c00e-4dad-9692-bbe1b6e93e37"
              onClick={[Function]}
            >
              Delete
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
